src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,186 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    public static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> previous;
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.previous = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
+    private Node<T> firstNode;
+    private Node<T> lastNode;
```

Comment:
In this data structure, the first node would be called "head" and the last "tail"
variable names should reflect this